web/src/components/users/AdminUserEditForm.vue

```diff
@@ -0,0 +1,307 @@
+<template>
+  <v-skeleton-loader
+    v-if="isNil(user)"
+    type="card"
+  />
+  <v-form
+    v-else
+    v-model="isValid"
+    @submit.prevent="saveWrapper"
+  >
+    <v-row>
+      <v-col
+        cols="12"
+        md="6"
+      >
+        <v-text-field
+          v-model="user.firstName"
+          label="First name *"
+          :rules="[required]"
+          variant="outlined"
+          required
+        />
+      </v-col>
+      <v-col
+        cols="12"
+        md="6"
+      >
+        <v-text-field
+          v-model="user.lastName"
+          label="Last name *"
+          :rules="[required]"
+          variant="outlined"
+          required
+        />
+      </v-col>
+    </v-row>
+
+    <v-row>
+      <v-col
+        cols="12"
+        md="6"
+      >
+        <v-text-field
+          v-model="user.email"
+          label="Email *"
+          :rules="[required]"
+          variant="outlined"
+          required
+        />
+      </v-col>
+      <v-col
+        cols="12"
+        md="6"
+      >
+        <v-text-field
+          v-model="user.position"
+          label="Position"
+          variant="outlined"
+        />
+      </v-col>
+    </v-row>
+
+    <v-divider class="my-6"></v-divider>
+
+    <v-row>
+      <v-col
+        cols="12"
+        md="6"
+      >
+        <UserGroupAutocomplete
+          :model-value="groupMembershipAttributes.departmentId"
+          :type="UserGroupTypes.DEPARTMENT"
+          :parent-id="null"
+          :rules="[required]"
+          label="Department"
+          variant="outlined"
+          required
+          @update:model-value="updateDepartment"
+        />
+      </v-col>
+      <v-col
+        cols="12"
+        md="6"
+      >
+        <UserGroupAutocomplete
+          :model-value="groupMembershipAttributes.divisionId"
+          :type="UserGroupTypes.DIVISION"
+          :parent-id="groupMembershipAttributes.departmentId"
+          label="Division"
+          variant="outlined"
+          clearable
+          :readonly="isNil(groupMembershipAttributes.departmentId)"
+          :append-inner-icon="isNil(groupMembershipAttributes.departmentId) ? 'mdi-lock' : ''"
```

Comment:
So ... this is a case where I would want to use `disabled`.
Conceptually, the field only activates once you fill the parent.

I see "locking" a field as something you would do when displaying a never editable field (e.g. email) in the middle of a bunch of editable fields (e.g. first name, last name)

Reply:
My previous request https://github.com/icefoganalytics/internal-data-portal/pull/116#discussion_r1801560991, is a bit confusing in contrast.

In that version, you where showing a field to a user for display purposes, but making it hard for them to read, via disabled, so I recommended the "locked" instead of "disabled" view. As that is easier to ready in a display context.

In this version, the field is only _temporarily_ inaccessible until you fill in parent values; as such, it makes sense to make the field hard to read/easy to ignore, until the parent is filled in. At which point the field becomes "non-disabled" and easy to read, drawing the eye, and encouraging the user to then act on it.

Reply:
The fact the the UI now shows a different view to the User and Admin makes it possible to make use of this kind of UI (UX?) enhancement.